### PR TITLE
PLT-4515 Change direct message list to be infinite scroll

### PR DESCRIPTION
**Pull request by dmeza**
#### Summary
- @mfpiccolo Added infinite scrolling to Direct Messages>More..>Members list.
- Changed some of the styles to display search on a single line and reduce used paddings for header and footer.
- Sized scrolling container based on the window height so that mobile devices display full  
#### Ticket Link

https://mattermost.atlassian.net/browse/PLT-4515
#### Checklist

[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] Added or updated unit tests (required for all new features)
- [ ] Added API documentation (required for all new APIs)
- [ ] All new/modified APIs include changes to the drivers
- [ ] Has enterprise changes (please link)
- [x] Has UI changes
- [ ] Includes text changes and localization file
- [ ] Touches critical sections of the codebase (auth, upgrade, etc.)

### CODE REVIEW

File: webapp/components/searchable_user_list.jsx [hunk 1]
```diff
@@ -147,15 +161,41 @@ export default class SearchableUserList extends React.Component {
...
+
+            if (this.props.infinite) {
+                if (this.state.nextDisabled) {
+                    pageNavLinks = (
+                        <div className='more-modal__loading-bar'>
+                            {'Loading...'}
+                        </div>
+                    );
+                } else {
+                    pageNavLinks = (
```
Comment (mfpiccolo) [inline webapp/components/searchable_user_list.jsx]: We can probably remove or refactor this bit.  The Not Loading was meant to be a spike as I was developing this.

File: webapp/components/searchable_user_list.jsx [hunk 2]
```diff
@@ -147,15 +161,41 @@ export default class SearchableUserList extends React.Component {
...
+            }
         }
 
+        const height = $(window).height() - ($(window).width() <= 768 ? 95 : 198);
+
         return (
             <div
                 className='filtered-user-list'
-                style={this.props.style}
+                style={$.extend(this.props.style, {'max-height': `${height}px`})}
```
Comment (mfpiccolo) [inline webapp/components/searchable_user_list.jsx]: Can we use es6 syntax here?

```
{...this.props.style, max-height: `${height}px`}
```

**Comment by dmeza** [comment]
@mfpiccolo @coreyhulen errors fixed.

**Comment by mfpiccolo** [comment]
It is also worth noting that this branch has an issue with IOS inertial scrolling.

**Comment by jasonblais** [comment]
Thanks @dmeza, @mfpiccolo,

Tested and works great on Windows/Chrome. The loading of new users happens quite elegantly on the browser

image

On iPhone 5s, iOS 10: As I scroll through the direct message list, I sometimes get a white screen as it loads more users. Wondering if there was a way to fix that and make the user experience smoother?

**Comment by jasonblais** [comment]
@dmeza , @mfpiccolo 

Found a second issue:
1) Open DM more list
2) Scroll through user list, then hit "Close"
3) Re-open the DM more list, and scroll through the list again

Observed: After re-opening the DM list, the app let's me load more users only once. Reproduced a few times on [the testing server](http://i-9c8a4dc3.spinmint.com:8065/pr4375/channels/town-square#) (Windows, Chrome)

**Comment by mfpiccolo** [comment]
@jasonblais RE: issue # 1. 

The issue you are experiencing on IOS is due to momentum scrolling.  We have not yet found a simple solution for this.  There is a semi solution built into react-infinite, however it basically loads more elements into the dom and really only makes the problem happen less often but does not actually fix it.  I imagine that we would need to fork react-infinite and fix it inside this component.  Something along the lines of changing how the component is listening to scroll events to take into account inertail/momentum scrolling.

**Comment by jasonblais** [comment]
Thanks @mfpiccolo for the reply!

I'm going to check with Shift and hear there thoughts -- I worry that the user experience will be worse on mobile than with the existing direct message list

**Comment by jasonblais** [comment]
Chatted with @thomchop , with two days past feature complete, and with significant work to get this right, we won't be able to take infinite scrolling in 3.5. There's just too much risk involved.

Have forwarded information to Shift regarding potential next steps.

**Comment by jasonblais** [comment]
Closing this PR for now - let's reconsider it when we start working on the React Native apps.

**Pull request closed without merging**